<!DOCTYPE html>
<HTML>
	<HEAD>
		<TITLE> KU teaches RNA-seq </TITLE>
		<LINK REL = "stylesheet" TYPE = "text/css" HREF = "style2.css">
		<script>
			function toggleExtra(x) {
            console.log("toggleExtra("+x+")")
			document.getElementById("extra"+x).classList.toggle("show");
			};
		</script>
	</HEAD>
	<body>
		<div id="container">
		<div id="wrapper">
		<h1>HTseq</h1>
		<div id="page">
        <div id="content">
            <div class="box" id="box1">
                <h3>Summary
                </h3>
                <!--
                1. Simple explanation of what it does (visuals?)
                    what kind of information does this software use?
                        what does it mean?
                -->
                <div class="infobox indentme">
                    <!-- summary of what the program does. Perhaps links to the subject matter -->
                    counts are generated by finding the number of reads the overlap with a gene from our gene model.
                    <a href="https://www.ebi.ac.uk/training/online/course/functional-genomics-ii-common-technologies-and-data-analysis-methods/quantification" class="hiddenlink">[Still Confused?]</a>
                </div>
            </div>
            <div class="box" id="box2">
                <!--
                2. how to install program
                -->
                <h3>
                    <span class="clickable1" id="togglebid5" onclick="toggleExtra('bid5')">
                        Installation
                    </span>
                </h3>
                <div class="extra indentme" id="extrabid5">
                     For linux<br>
                     <ol start="1">
                         <li>
                             You must first install <a href="https://pypi.org/project/pip/">Pip</a> before continuing.
                         </li>
                         <li>
                             <code>pip install HTSeq</code><br>
                         </li>
                     </ol>
                    <a href="https://htseq.readthedocs.io/en/release_0.10.0/install.html#installation-on-linux" class="hiddenlink">[Installation Guide]</a><br>
                </div>
            </div>
            <div class="box" id="box3">
                <h3>How to use this program
                </h3>
                <!--
                3. examples showing off how to use the program
                    3.1 sample files students can feed into the program
                    3.2 instructions
                    3.3 output to expect (so they know if they're using it correctly)
                -->
                <h4>usage</h4>
                <div class="infobox indentme">
                    <code>htseq-count [options] &lt;alignment_files&gt; &lt;gff_file&gt;</code><br>
                    <a href="read further url:http://htseq.readthedocs.io/en/master/count.html" class="hiddenlink">[Read Further]</a><br>
                    <br>
                </div>
                
                <h4>my command</h4>
                <div class="infobox indentme">
                    <div class="codebox">
                    htseq-count -s no -r pos -t exon -i Parent -f bam $RNApath/map/SRR1930186Aligned.sortedByCoord.out.bam \<br>
                    $RNApath/genome/annotations/Cyanidioschyzon_merolae.ASM9120v1.39.chromosome.19.gff3 &gt; $RNApath/counts/SRR1930186-output_basename.counts<br>
                    htseq-count -s no -r pos -t exon -i Parent -f bam $RNApath/map/SRR1930185Aligned.sortedByCoord.out.bam \<br>
                    $RNApath/genome/annotations/Cyanidioschyzon_merolae.ASM9120v1.39.chromosome.19.gff3 &gt; $RNApath/counts/SRR1930185-output_basename.counts<br>
                    htseq-count -s no -r pos -t exon -i Parent -f bam $RNApath/map/SRR1930184Aligned.sortedByCoord.out.bam \<br>
                    $RNApath/genome/annotations/Cyanidioschyzon_merolae.ASM9120v1.39.chromosome.19.gff3 &gt; $RNApath/counts/SRR1930184-output_basename.counts<br>
                    htseq-count -s no -r pos -t exon -i Parent -f bam $RNApath/map/SRR1930183Aligned.sortedByCoord.out.bam \<br>
                    $RNApath/genome/annotations/Cyanidioschyzon_merolae.ASM9120v1.39.chromosome.19.gff3 &gt; $RNApath/counts/SRR1930183-output_basename.counts
                    </div><br>
                </div>

            </div>
            <a href="TophatOverview.html">
                <div class="navbut left">Tophat</div>
            </a>
            <a href="BEDtoolsOverview.html">
                <div class="navbut right">BEDtools</div>
            </a>
        </div>
		</div>
        </div>
        </div>
	</body>
</html>
